framework/include/utils/PerfGraphRegistry.h

```diff
@@ -46,26 +46,51 @@ class PerfGraphRegistry
   class SectionInfo
   {
   public:
-    SectionInfo(
-        PerfID id, std::string name, unsigned int level, std::string live_message, bool print_dots)
+    SectionInfo(const PerfID id,
+                const std::string & name,
+                const unsigned int level,
+                const std::string & live_message,
+                const bool print_dots)
       : _id(id), _name(name), _level(level), _live_message(live_message), _print_dots(print_dots)
     {
     }
 
+    /**
+     * @returns The unique ID for the section
+     */
+    PerfID id() const { return _id; }
+    /**
+     * @returns The name of the section
+     */
+    const std::string & name() const { return _name; }
+    /**
+     * @returns The verbosity level of the section
+     */
+    unsigned int level() const { return _level; }
+    /**
+     * @returns The message to print while the section is running
+     */
+    const std::string & liveMessage() const { return _live_message; }
+    /**
+     * @returns Whether or not to print dots while this section runs
+     */
+    bool printDots() const { return _print_dots; }
+
```

Comment:
Personally - I think this is pretty useless.  This class was originally a `struct`... but it needed initialization so I made it a class just to get the constructor.

Note - it only needed the constructor because of a limitation in C++11 that will be fixed in later versions of C++ (structs can't currently be initialized with initialization syntax)... so this constructor could be removed later and and this class get go back to being a struct.

Reply:
I made everything `const` so that I can make `PerfNodeInfo` `const`-correct.

Adding the methods is just extraneous so that the variables can be private. I can remove the methods if you prefer.

Reply:
I just don't think they add anything... but it doesn't matter much either way.